### PR TITLE
Add OMNeT++ extension

### PR DESCRIPTION
**Pull request by adamgeorge309**
NED and MSG (extensions .ned and .msg) are DSLs for the OMNeT++ network simulator. They are used by model frameworks for OMNeT++, such as the INET framework, Simu5G, Veins, etc. We are the team that develops [OMNeT++](https://github.com/omnetpp/omnetpp) and the [INET Framework](https://github.com/inet-framework/) and we have created tree-sitter grammars for these languages, as well as an extension for the Zed editor.

### CODE REVIEW

**Comment by adamgeorge309** [comment]
@cla-bot check

**Review by maxdeviant — CHANGES_REQUESTED** [review]
Could you please add the submodule to this PR?

The `.gitmodules` entry is present, but the submodule itself is not.

File: extensions.toml [hunk 1]
```diff
@@ -794,6 +794,11 @@ version = "0.1.0"
 submodule = "extensions/odin"
 version = "0.1.0"
 
+[omnetpp]
+submodule = "extensions/omnetpp-zed"
+path = "extensions/omnetpp"
```
Comment (maxdeviant) [inline extensions.toml:799]: We don't need this `path` here, as the extension repo has the extension at the root.

```suggestion
```

**Comment by adamgeorge309** [comment]
I added the submodule and accepted your change. Thanks!

File: extensions/omnetpp-zed
Comment (maxdeviant) [inline extensions/omnetpp-zed:1]: Can we change this directory to be just `extensions/omnetpp`, so it matches the extension ID?

**Comment by maxdeviant** [comment]
It looks like there may be some conflicts with `main` that need to be resolved.

File: extensions/omnetpp-zed
Comment (adamgeorge309) [reply extensions/omnetpp-zed:1]: Yes, we meant to do that, so that we don't have zed in the name.

Comment (adamgeorge309) [reply extensions/omnetpp-zed:1]: Do I understand correctly that `path = extensions/omnetpp` in .gitmodules specifies the directory name?

Comment (maxdeviant) [reply extensions/omnetpp-zed:1]: `path` is for a path within the extension repository. It's generally not needed unless you have multiple extensions in a repository, or need them located someone other than the root.

But what we want is for the directory in this repository to be `extensions/omnetpp`. The actual repository name can be `omnetpp-zed`.

What you want to do is add the submodule like this:

```sh
git submodule add https://github.com/omnetpp/omnetpp-zed.git extensions/omnetpp
```

Comment (adamgeorge309) [reply extensions/omnetpp-zed:1]: I did this, thank you!

**Comment by adamgeorge309** [comment]
I fixed the extension ID in extension.toml to match extensions.toml

**Comment by maxdeviant** [comment]
> @maxdeviant, I fixed the extension ID in extension.toml to match extensions.toml

@adamgeorge309 I see the correct ID in your extension repository, but the submodule commit in this repo needs to be updated to match. I don't have the ability to push any changes to this branch, as your fork is on a GitHub organization.

**Comment by adamgeorge309** [comment]
Thank you

**Pull request merged**